public/video-ui/src/actions/VideoActions/saveVideo.js

```diff
@@ -45,24 +45,29 @@ export function saveVideo(video) {
   return dispatch => {
     dispatch(requestVideoSave(video));
     return VideosApi.saveVideo(video.id, video)
-    .then(res => {
-      dispatch(receiveVideoSave(res))
-      return VideosApi.getVideoUsages(video.id)
-    })
-    .then(usages => {
-      dispatch(receiveVideoUsages(usages));
-      return VideosApi.updateCanonicalPages(
-        video,
-        usages,
-        'preview'
-      )
-      .then(() => dispatch(receiveVideoPageUpdate(video.title)))
-    })
-    .catch(error => {
-      const defaultError = 'Could not save video';
-      const conflictError = 'Could not save video as another user (or tab) is currently editing this video';
-      const message = error.status === 409 ? conflictError : defaultError;
-      dispatch(errorVideoSave(error, message));
-    });
+      .then(res => {
+        dispatch(receiveVideoSave(res));
+        return VideosApi.getVideoUsages(video.id)
+          .then(usages => {
+            dispatch(receiveVideoUsages(usages));
+            return VideosApi.updateCanonicalPages(
+              video,
+              usages,
+              'preview'
+            );
+          })
+          .then(() => dispatch(receiveVideoPageUpdate(video.title)))
+          .catch(error => {
+            const message = 'Could not update canonical page';
+            dispatch(errorVideoSave(error, message));
+          });
+      })
+      .catch(error => {
+        const defaultError = 'Could not save video';
+        const conflictError = 'Could not save video as another user (or tab) is currently editing this video';
+        const message = error.status === 409 ? conflictError : defaultError;
+        dispatch(errorVideoSave(error, message));
+        throw error;
+      });
```

Comment:
I do have a concern that allowing a dispatching function to throw errors might break other usages - if I'm understanding the redux correctly, components like `PlutoProjectPicker` have this `saveVideo` passed down to them as `props.saveVideo` via `mapDispatchToProps` (for `PlutoProjectPicker`, the prop comes from its parent, `VideoUpload` ), but they don't allow for it throwing like `VideoDisplay` now does.

This wouldn't be ideal, but we could add an optional "allowThrowing" argument to `saveVideo` so it could be made to only throw the error with calling by a component that allows for it?